src/components/teams.js

```diff
@@ -53,16 +56,29 @@ export default React.createClass({
           renderRow={this.renderRow}
         />
         <View style={styles.footer}>
-          <Button onPress={this.props.addTeam}>Add Team</Button>
+          <Button onPress={this.goToAddTeam}>Add Team</Button>
         </View>
       </View>
     )
   },
+  goToAddTeam() {
+    this.props.navigator.push({
+      title: 'Add Team',
+      component: AddTeam,
+      passProps: {
+        onComplete: this.onCompleteAddTeam
+      },
+    });
+  },
   renderRow(teamRef, sectionID, rowID) {
     return (
-      <ListViewItem key={teamRef.key()} teamRef={teamRef}></ListViewItem>
+      <ListViewItem key={teamRef.key()} teamRef={teamRef} onPress={this.props.onPress}></ListViewItem>
     )
   },
+
+  onCompleteAddTeam(teamName) {
+    this.props.navigator.pop();
+  },
```

Comment:
See how navigation concerns get spread throughout your code. This is another reason why routers are awesome - you can control your state transitions in a single, separate layer of your system.